examples/map/map_segment.py

```diff
@@ -0,0 +1,98 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+This example demonstrates extracting a region of a particular map based on
+world coordinates in different systems.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+######################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+######################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel. From this we then transform the coordinates to the required frame.
+# For this example we are going to extract a region based on the
+# heliographic Stonyhurst coordinates, so we transform to that frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+######################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where `True` indicates invalid or deselected data.
+# mask. Numpy's masked arrays allow for a combination of standard numpy array and
+# a mask array. When an element of the mask is `True`, the corresponding element
+# of the associated array is said to be masked (invalid).
+# `More information about numpy's masked arrays <https://numpy.org/doc/stable/reference/maskedarray.generic.html#using-numpy-ma>`
+# From the segment, we now mask out all values where ``all_hgs.lon`` is NaN.
+# This is the convention used by the `numpy.ma` module.
+segment_mask = np.logical_or(all_hgs.lon >= 35 * u.deg, all_hgs.lon <= -35 * u.deg)
+segment_mask |= np.isnan(all_hgs.lon)
+
+######################################################################
+# Let's plot the segment separately, we create a new map with the segment as the mask
+new_frame_map = sunpy.map.Map(smap.data, smap.meta, mask=segment_mask)
+fig = plt.figure()
+ax = plt.subplot(projection=smap)
+new_frame_map.plot()
+new_frame_map.draw_grid(color='red')
+plt.show()
+
+######################################################################
+# We can perform various mathematical operations on the extracted segment such
+# as averaging the pixel values or finding the sum of the segment.
+masked_data = np.ma.array(smap.data, mask=segment_mask)
+
+print(f"Original Map : mean = {smap.data.mean()}, sum = {smap.data.sum()}")
+print(f"Segment : mean = {masked_data.mean()}, sum = {masked_data.sum()}")
+
+######################################################################
+# This example can also be extended while using `sunpy.coordinates.NorthOffsetFrame`
```

Comment:
I think this will work and create a subheading.
```suggestion
# Using `sunpy.coordinates.NorthOffsetFrame`
# ------------------------------------------
```